File: DZ03A/src/hr/unipu/java/Main.java
package hr.unipu.java;
import java.util.Scanner;

/**
 *
 * @author Pritišanac Tin
 * @version 1.0
 * DZ 3A korištenjem kompozicije
 * Kod svih zadaća dostupan je na git repozitoriju: https://github.com/AlphaActual/JavaVirtualMachine
 *
 */

public class Main {

    public static void main(String[] args) {

        // Instanciranje novog kviza
        Quiz quiz = null;

        do {
            Quiz.showMenu();
            Scanner scanner = new Scanner(System.in);
            if(!isInputInt(scanner)) {
                continue;
            }
            int choice = scanner.nextInt(); scanner.nextLine();
            switch (choice) {
                // dodavanje pitanja kompozicijom
                case 1:
                    // Postavljanje novog kviza
                    if(quiz == null) {
                        System.out.println("Unesite ime kviza: ");
                        String name = scanner.nextLine();
                        System.out.println("Unesite maksimalan broj pitanja: ");
                        int maxQuestions = scanner.nextInt(); scanner.nextLine();
                        quiz = new Quiz(name, maxQuestions);
                    }
                    if(quiz.questionsFull()) {
                        System.out.println("Nije moguće dodati više pitanja!");
                        break;
                    }
                    System.out.println("Unesite pitanje: ");
                    String question = scanner.nextLine();
                    System.out.println("Unesite točan odgovor: ");
                    String answer = scanner.nextLine();
                    boolean questionAdded = quiz.addQuestion(question, answer);
                    if(questionAdded) {
                        System.out.println("Pitanje je uspješno dodano!");
                    }
                    break;
                case 2:
                    if (quiz == null) {
                        System.out.println("Nije moguće pokrenuti kviz jer nije postavljen niti jedan kviz!");
                        break;
                    }
                    quiz.runQuiz();
                    break;
                case 3:
                    System.exit(0);
                    break;
                default:
                    System.out.println("Nepostojeći izbor!");
            }
        } while (true);

    }
    public static boolean isInputInt(Scanner scanner) {
        if(!scanner.hasNextInt()){
            System.out.println("Netočan unos. Unesite brojčanu vrijednost");
            scanner.next(); // consume the invalid input
            return false;
        }
        return true;
    }
}

File: DZ03A/src/hr/unipu/kotlin/Main.kt
package hr.unipu.kotlin

import java.util.Scanner

fun main(args: Array<String>) {

    // Instanciranje novog kviza
    val quiz = Quiz()

    // Dodavanje pitanja i točnih odgovora (kompozicija)
    quiz.addQuestion("U kojem jeziku se programira za Android?", "Java")
    quiz.addQuestion("U kojem jeziku se programira za iPhone?", "Objective-C")

    // Dodavanje pitanja i točnih odgovora (agregacija)
    //    val question1 = ShortAnswerQuestion("U kojem jeziku se programira za Android?", "Java")
    //    quiz.addQuestion(question1)
    //    val question2 = ShortAnswerQuestion("U kojem jeziku se programira za iPhone?", "Objective-C")
    //    quiz.addQuestion(question2)


    // Učitavanje odgovora s tipkovnice
    val scanner = Scanner(System.`in`)
    var noOfCorrectAnswers = 0
    for (i in 0 until quiz.noOfQuestions) {
        println(quiz.getQuestion(i))
        val answer = scanner.nextLine()
        if (quiz.isCorrectAnswer(i, answer)!!) {
            noOfCorrectAnswers++
            println("Točno!")
        } else {
            println("Netočno.")
        }
    }

    println("Imali ste " + noOfCorrectAnswers + " točnih odgovora od ukupno " + quiz.noOfQuestions + ".")

}
